Sources/Utility/Process.swift

```diff
@@ -165,6 +171,21 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    func validateArgumentIsExecutable(_ program: String) throws {
+
+        // Check to see if the argument is already listed in the vaildated cache.
+        guard !Process.validatedExecutables.contains(program) else {
+            return
+        }
+        let envSearchPaths = Utility.getEnvSearchPaths(pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
```

Comment:
I haven't tried compiling this but we need to do something similar to this:
```swift
    /// Queue to protect reading/writing on executables map.
    private static let executablesQueue = DispatchQueue(
            label: "org.swift.swiftpm.process.findExecutable")

    /// Contains results of executable lookup.
    ///
    /// Key: Executable name or path.
    /// Value: If key was found in the search paths.
    static private var executablesMap: [String: Bool] = [:]

    /// Returns true if the given program is present in search path.
    ///
    /// The program can be executable name, relative path or absolute path.
    func findExecutable(_ program: String) -> Bool {
        return Process.executablesQueue.sync {
            // Check if we already have a value for the program.
            if let value = Process.executablesMap[program] {
                return value
            }
            // FIXME: This can be cached.
            let envSearchPaths = Utility.getEnvSearchPaths(
                pathString: getenv("PATH"),
                currentWorkingDirectory: currentWorkingDirectory
            )
            // Lookup the executable.
            let value = Utility.lookupExecutablePath(
                filename: program, searchPaths: envSearchPaths) != nil
            Process.executablesMap[program] = value
            return value
        }
    }

```